scripts/issue/1133/upkeep_manager_integration.py

```diff
@@ -13,27 +13,101 @@
+    "GasStationExact": 89,
 }
 
+techops = GreatApeSafe(r.badger_wallets.techops_multisig)
+upkeep_manager_snap = GreatApeSafe(r.badger_wallets.upkeep_manager)
 
-def main():
-    techops = GreatApeSafe(r.badger_wallets.techops_multisig)
+# contract
+upkeep_manager = techops.contract(r.badger_wallets.upkeep_manager)
+
+
+def initial_setup():
+    """
+    carries the following actions: cancels upkeeps from both registries,
+    sends techops link funds to manager, init the base upkeep in the manager,
+    updates watchlist, updates registry used in the gas station to latest and
+    register the gas station so it sends 1 eth to the manager
+    """
     techops.init_badger()
+    techops.init_chainlink()
+
+    # snaps
+    techops.take_snapshot([techops.chainlink.link])
+    upkeep_manager_snap.take_snapshot([techops.chainlink.link])
+
+    initial_rounds_top_up = upkeep_manager.roundsTopUp()
+
+    # 1. Send links funds to allow to init base upkeep job
+    link_bal = techops.chainlink.link.balanceOf(techops)
+    techops.chainlink.link.transfer(upkeep_manager, link_bal)
 
-    # contracts involved
-    cl_registry = techops.contract(
-        r.chainlink.keeper_registry, interface.IKeeperRegistry
-    )
-    upkeep_manager = techops.contract(r.badger_wallets.upkeep_manager)
+    # NOTE: reduce the min rounds so we can initialize it, since techops only has 90 LINK
+    upkeep_manager.setRoundsTopUp(2)
 
-    # 1. Add existing members
+    # NOTE: while testing with foundry, setting up on this figure have not seen any revert
+    # on top-up actions or withdrawal of funds from the keepers.
+    # ref of gas snap: https://github.com/Badger-Finance/badger-avatars/blob/main/.gas-snapshot#L166-L176
+    upkeep_manager.initializeBaseUpkeep(1_000_000)
+
+    upkeep_manager_snap.print_snapshot()
+
+    # 2. Cancel upkeep in old and new registry to migrate ownership to the upkeep manager
     for name, upkeep_id in members.items():
-        member_address, gas_limit, _, _, _, _, _, _ = cl_registry.getUpkeep(upkeep_id)
-        # https://etherscan.io/address/0x4c02f0160dc0387b13bcb5e1728c780649e109ac#code#F16#L166
-        upkeep_manager.addMember(member_address, name, gas_limit, upkeep_id)
+        if name == "GasStationExact":
+            techops.chainlink.keeper_registry_v1_1.cancelUpkeep(upkeep_id)
+        else:
+            techops.chainlink.keeper_registry.cancelUpkeep(upkeep_id)
 
-    # 2. Include manager in gas st watchlist
+    # 3. Update gas station
     techops.badger.set_gas_station_watchlist()
 
+    # 4. Set `s_keeperRegistryAddress` to latest cl registry
+    techops.badger.station.setKeeperRegistryAddress(techops.chainlink.keeper_registry)
+
+    # 5. Add gas st in this tx with the left-over link funds to auto-fund the manager in next block with eth
+    upkeep_manager.addMember(techops.badger.station, "GasStationExact", 400_000, 0)
+    upkeep_manager.setRoundsTopUp(initial_rounds_top_up)
+
+    upkeep_manager_snap.print_snapshot()
```

Comment:
if already adding the gas station here, at what point do you plan to add the dummy upkeep for testing?

i think it would make more sense to
- add a dummy here instead of the gas station
- manager can be funded from a deployer with some eth
- register all members at once (as soon as dummy run looks successful)

Reply:
they're separate testing, since we cannot enforce easily an under-funded upkeep afaik, here we add the gas st to integrate the flow of sending the eth to the upkeep manager, having the live-test of one real upkeep task done

while the dummy one it is to test the flow of the keepers adding a member, later cancelling it and see how the keeper in auto retrieve the links funds for that specific upkeep

Reply:
as i said on our call, i think we should test with a dummy contract first before involving live, operational contracts.

Reply:
i reorg the script a bit better to align what we discussed in the calls in 4d52798ffae43071294ed10b16f28a0e0e8f5161, together with a sim, which walks thru the whole flow at once